pulse2percept/percepts/base.py

```diff
@@ -396,6 +400,9 @@ def save(self, fname, shape=None, fps=None):
                     if h % VIDEO_BLOCK_SIZE > 0:
                         out_h += VIDEO_BLOCK_SIZE - (h % VIDEO_BLOCK_SIZE)
                     data = resize(data, (out_h, out_w))
-            data = img_as_uint(data).astype(np.uint8)
-            imageio.mimwrite(fname, data.transpose((2, 0, 1)), fps=fps)
+            data = img_as_ubyte(data)
+            try:
+                imageio.mimwrite(fname, data.transpose((2, 0, 1)), fps=fps)
+            except:
```

Comment:
Do we know what error is thrown here? Would be better to either only catch the right error, or ideally, do the switch beforehand based on when its necessary (guessing its determined by filetype).

Reply:
you're right -- just pushed something to ensure that we only catch TypeError